ci/functional/test_main_prep_node.sh

```diff
@@ -2,33 +2,226 @@
+        echo "and"
+        echo "$opa_count Omni-Path adapters"
+        echo "The Onmi-Path adapters will not be used."
+        })"
+    mail_message+="${nl}${ib_message}${nl}"
+    echo "$ib_message"
+fi
+set -x
+
+function do_wait_for_ib {
+    ib_timeout=300 # 5 minutes
+    retry_wait=10 # seconds
+    timeout=$((SECONDS + ib_timeout))
+    while [ "$SECONDS" -lt "$timeout" ]; do
+      ip link set up "$1" || true
+      sleep 2
+      if ip addr show "$iface" | grep "inet "; then
+        return 0
+      fi
+      sleep ${retry_wait}
+    done
+}
+
+# First check for infinband devices
+for i in $(seq 0 $ib_count); do
     iface="ib$i"
-    if [ -e /sys/class/net/"$iface" ]; then
-        if ! ifconfig "$iface" | grep "inet "; then
-          {
-            echo "Found interface $iface down after reboot on $HOSTNAME"
-            systemctl status || true
-            systemctl --failed || true
-            journalctl -n 500 || true
-            ifconfig "$iface" || true
-            cat /sys/class/net/"$iface"/mode || true
-            ifup "$iface" || true
-            ifconfig -a || true
-            ls -l /etc/sysconfig/network-scripts/ifcfg-* || true
-            cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
-          } 2>&1 | mail -s "Interface $iface found down after reboot" \
-                   -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL"
+    if do_wait_for_ib "$iface"; then
+        set +x
+        if ! ip addr show "$iface" | grep "inet "; then
+            ib_message="$({
+                echo "Found interface $iface down after reboot on $HOSTNAME"
+                ip addr show "$iface" || true
+                cat /sys/class/net/"$iface"/mode || true
+                ip link set up "$iface" || true
+                cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
+                } 2>&1)"
+            mail_message+="${nl}${ib_message}${nl}"
+            echo "$ib_message"
+        else
+            echo "OK: Interface $iface is up."
         fi
-        if ! ifconfig "$iface" | grep "inet "; then
+        if ! ip addr show "$iface" | grep "inet "; then
+            echo "Fail: $ib_message"
+            result=1
             echo "Failed to bring up interface $iface on $HOSTNAME. " \
-            "Please file a CORCI ticket."
-            exit 1
+            "Please file a SRE ticket."
         fi
+        if [ -e "/sys/class/net/$iface/device/numa_node" ]; then
+            set -x
+            cat "/sys/class/net/$iface/device/numa_node"
+        fi
+        set -x
+    else
+        ib_message="Failed to bring up interface $iface on $HOSTNAME. "
+        mail_message+="${nl}${ib_message}${nl}"
+        echo "$ib_message"
+        result=1
+        echo "Please file a SRE ticket."
     fi
 done
 
-if ! grep /mnt/share /proc/mounts; then
+# having -x just makes the console log harder to read.
+set +x
+if [ -e /sys/class/net/ib1 ]; then
+    # now check for pmem & NVMe drives when ib1 is present.
```

Comment:
We only use that hardware in CI tests on systems with 2 interfaces.
We will probably need a modification to this in the future for clusters with out pmem.

Reply:
We would need to pass to the tests or better yet to the provisioning process a complete list of what is needed on the provisioned cluster.   This would be needed if we could move to something like SLURM run the test jobs on a cluster it creates and then notify the caller that the job is done.

Reply:
But wouldn't a heuristic such as if no PMEM is detected at all, skip checking it (and assume PMEM is not needed for this test) but if any is found, check it?  What are the chances that PMEM is needed and every stick of it in a machine is completely undetectable?

Reply:
That would be a future check to be deployed.
Currently all our hardware CI with two IB adapters have PMEM and NVMe.
We do not have any of the hardware that will be replacing PMEM or IB for future configurations.
So we will have to add new tests for when that happens.

Longer term probably need to store an expected hardware configuration for each node where it can be checked cluster provisioning time.   This PR will at least get immediately get us 2 large and 1 medium clusters added for CI throughput, and eventually another medium cluster once I can get the nodes back from the developers.
A version of it will likely be needed for the 2.4 branch to prevent intermittent test failures.